src/PortfolioPlanning/Common/Services/BacklogConfigurationDataService.ts

```diff
@@ -30,12 +31,17 @@ export class BacklogConfigurationDataService {
 
         const projectEfforFieldRefName =
             projectBacklogConfiguration.backlogFields &&
-                projectBacklogConfiguration.backlogFields.typeFields[BacklogConfigurationDataService.EffortTypeField]
+            projectBacklogConfiguration.backlogFields.typeFields[BacklogConfigurationDataService.EffortTypeField]
                 ? projectBacklogConfiguration.backlogFields.typeFields[BacklogConfigurationDataService.EffortTypeField]
                 : null;
 
         const portfolioLevelsData = this.getPortfolioLevelsData(projectBacklogConfiguration);
 
+        const portfolioWorkItemTypes = Object.keys(portfolioLevelsData.backlogLevelNamesByWorkItemType);
+        const workItemTypeMappedStates: WorkItemTypeStateInfo[] = projectBacklogConfiguration.workItemTypeMappedStates.filter(
+            item => portfolioWorkItemTypes.indexOf(item.workItemTypeName.toLowerCase()) !== -1
+        );
```

Comment:
I'd put these in a separate function that will be called by `DependencyPanel` instead of adding it to the `ProjectBacklogConfiguration` contract... not everyone that requests `ProjectBacklogConfiguration` needs the state categories mapping, just the `DependencyPanel`...